_posts/2019-11-12-customizing-episerver-content-delivery-api.md

```diff
@@ -0,0 +1,927 @@
+---
+layout: post
+title: Customizing Episerver content delivery API.
+author: demonru
+excerpt: It was my first project with the use of Episerver as a Headless CMS. Episerver CMS supports such mode but requires some customization for the exact needs. 
+tags:
+- Episerver
+- Api
+- CMS
+- DOTNET
+---
+
+## Episerver as Headless CMS
+Episerver is a well known Content Management System (CMS). It officially supports working in a "headless CMS" mode. There is a bunch of [articles](https://world.episerver.com/documentation/developer-guides/content-delivery-api/) how to get started with it in this mode. In this article, I will be talking only about Content delivery API part of the CMS usage. Default Alloy template allows you to quickly get content API up and running, but the default output result is far from being perfect for real case usage and usually requires customization.
+
+To demonstrate the problem I'll start by adding `EPiServer.ContentDeliveryApi.Cms` dependency to default [Alloy site](https://world.episerver.com/documentation/developer-guides/CMS/getting-started/install-a-sample-site/). One gets predefined API endpoints and extensive output models with the default configuration. The endpoints support querying and searching content, authorization, can work with friendly URLs. 
```

Comment:
Maybe:
With the default configuration you get predefined API endpoints and extensive output models.

> can work with friendly URLs
"and works"